### ISSUE TITLE
LIST shouldn't assume a non-channel parameter is always an ELIST string 

### ISSUE THREAD

**Issue by emersonveenstra** [solanum-ircd/solanum#224]
Doing something like `/list channelname` without the leading `#` results in all the channels being returned, since there's no check in `mo_list` to make sure that a non-channel parameter is a valid ELIST string. This was fixed in -seven a while ago: https://github.com/freenode/ircd-seven/commit/6ea60b2297948211925e22bd1f284179d680b4ae

**Comment by dwfreed** [comment]
Agreed, please feel free to submit a PR porting that commit to solanum.
